src/client/Microsoft.Identity.Client/Instance/Discovery/RegionDiscoveryProvider.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Client.Instance.Discovery;
+using Microsoft.Identity.Client.Internal;
+
+namespace Microsoft.Identity.Client.Region
+{
+    internal class RegionDiscoveryProvider : IRegionDiscoveryProvider
```

Comment:
This logic already existing but was bundled with region auto-discovery so I split it

Reply:
much better.

---
In reply to: [602423987](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2509#discussion_r602423987) [](ancestors = 602423987)